src/Multivariate/ProductFun.jl

```diff
@@ -5,6 +5,16 @@
 
 export ProductFun
 
+## TODO:
+## In a newer version, an abstract type of ProducFun is needed, where different implementations are possible
+## however, refactoring this is a lot of effort...
+struct TensorIteratorFun{S<:UnivariateSpace, d, SS<:TensorSpace{NTuple{d, S}}, T<:Number} <: MultivariateFun{T, d}
+    space::SS
+    coefficients::AbstractVector{T} 
```

Comment:
Do we need `AbstractVector` as the type? Or should this be concretely typed instead?

Reply:
changed